### PR TITLE
arm: Remove TRUE/FALSE defines

### PR DESCRIPTION
**Pull request by lioncash**
- Removed the Debug parameter from ARMul_State since it isn't used.
- Changed ARMul_CoProInit to a void function. It always returned true.

### CODE REVIEW

File: src/core/arm/interpreter/arminit.cpp [hunk 1]
```diff
@@ -63,24 +63,22 @@ void ARMul_EmulateInit()
 \***************************************************************************/
 ARMul_State* ARMul_NewState(ARMul_State* state)
 {
-    unsigned i, j;
-
     memset (state, 0, sizeof (ARMul_State));
 
     state->Emulate = RUN;
-    for (i = 0; i < 16; i++) {
+    for (u8 i = 0; i < 16; i++) {
```
Comment (yuriks) [inline src/core/arm/interpreter/arminit.cpp]: Why u8?

Comment (lioncash) [inline src/core/arm/interpreter/arminit.cpp]: Because there's no need for a range higher than that.

Comment (yuriks) [inline src/core/arm/interpreter/arminit.cpp]: The explicit types are distracting and unecessary. (And if it mattered here, they might also be slower.)

Comment (lioncash) [inline src/core/arm/interpreter/arminit.cpp]: Whether or not they are distracting is completely relative, however I see your other point.

**Comment by bunnei** [comment]
LGTM

**Pull request merged**